File: src/main/scala/de/ailis/starglance/scene/Scene.scala
/*
 * $Id: JsCodeWriter.java 873 2009-05-24 16:12:59Z k $
 * Copyright (C) 2009 Klaus Reimer <k@ailis.de>
 * See LICENSE.txt file for licensing information.
 */

package de.ailis.starglance.math

import java.awt.Graphics2D

/**
 * A node in the scene.
 */

@serializable
class Scene
{
    /** The root node of the scene */
    var rootNode: Node

    def render(buffer: Graphics2D)
    {
        renderNode(rootNode, buffer, Matrix4d.identity)
    }

    private def renderNode(node: Node, buffer: Graphics2D, transform: Matrix4d)
    {
        
    }
}

File: threedee-collada/src/main/java/de/ailis/threedee/collada/entities/Scene.java
/*
 * Copyright (C) 2010 Klaus Reimer <k@ailis.de>
 * See LICENSE.txt for licensing information.
 */

package de.ailis.threedee.collada.entities;

import java.io.Serializable;


/**
 * A scene
 *
 * @author Klaus Reimer (k@ailis.de)
 */

public class Scene implements Serializable
{
    /** Serial version UID */
    private static final long serialVersionUID = 1;

    /** The instance visual scene */
    private InstanceVisualScene instanceVisualScene;


    /**
     * Returns the instance visual scene.
     *
     * @return The instance visual scene. May return empty if not specified
     */

    public InstanceVisualScene getInstanceVisualScene()
    {
        return this.instanceVisualScene;
    }


    /**
     * Sets the instance visual scene.
     *
     * @param instanceVisualScene
     *            The instance visual scene to set. Null for unspecified
     */

    public void setInstanceVisualScene(
            final InstanceVisualScene instanceVisualScene)
    {
        this.instanceVisualScene = instanceVisualScene;
    }
}
